Testing all the possible GETs the app will have to get

tanadendere opened the pull request:
This PR introduces the api fetches and the interfaces that were created to create types from the API responses.

tanadendere committed:
feat: :sparkles: Performing all possible fetches

Wrote fetches for all the possible GETs the app will have to make

tanadendere committed:
feat: :construction: Removed console.logs and changed type any to unknown

Removed console.logs and changed type any to unknown

tanadendere committed:
feat: :twisted_rightwards_arrows: Merge branch 'main' into utilities/api-calls

yanevdw commented on src/models/chapter.d.ts:
```diff
@@ -0,0 +1,143 @@
+  lang: string;
+  id: number;
+  href: string;
+}
```
For these ```unknown``` values, if they are optional, you could do ```vol?: string``` for example where the string refers to the type of the parameter if it is not null.

yanevdw commented on src/models/chapter.d.ts:
```diff
@@ -0,0 +1,143 @@
+
+export interface Chapter2 {
+  chap: string;
+  vol: unknown;
```
```Line 116```: Same here

yanevdw commented on src/models/comicChapters.d.ts:
```diff
@@ -0,0 +1,31 @@
+  id: number;
+  chap: string;
+  title: unknown;
+  vol: unknown;
```
Same here.

yanevdw commented on src/models/comicChapters.d.ts:
```diff
@@ -0,0 +1,31 @@
+  is_the_last_chapter: boolean;
+  group_name: string[];
+  hid: string;
+  identities: unknown;
+  md_chapters_groups: MdChaptersGroup[];
```
Here as well.

yanevdw commented on src/models/comicInfo.d.ts:
```diff
@@ -0,0 +1,165 @@
+  authors: Author[];
+  langList: string[];
+  recommendable: boolean;
+  demographic: unknown;
+  englishLink: unknown;
```
Here as well.

yanevdw commented on src/models/comicInfo.d.ts:
```diff
@@ -0,0 +1,165 @@
+  hid: string;
+  lang: string;
+  group_name: string[];
+  vol: unknown;
```
Same here.

yanevdw commented on src/models/comicInfo.d.ts:
```diff
@@ -0,0 +1,165 @@
+  links: Links;
+  last_chapter: number;
+  chapter_count: number;
+  demographic: unknown;
```
Here too.

yanevdw commented on src/models/comicInfo.d.ts:
```diff
@@ -0,0 +1,165 @@
+  content_rating: string;
+  translation_completed: boolean;
+  chapter_numbers_reset_on_new_volume_manual: boolean;
+  final_chapter: unknown;
+  final_volume: unknown;
```
Here too.

yanevdw reviewed (changes requested):
Well done! Just check all the unknowns in you models, if they are optional, rather define them as optional.

tanadendere committed:
feat: :zap: Changed the unknown types to optional

yanevdw reviewed (approved):
Changes have been implemented. Nicely done!

LesediMoko commented on src/models/chapter.d.ts:
```diff
@@ -0,0 +1,143 @@
+  href: string;
+}
+
+export interface Chapter2 {
```
Consider using the same type for Next and Chapter2. They have plenty of properties in common and those that are missing in one or the other can just be optional

LesediMoko commented on src/models/searchResult.d.ts:
```diff
@@ -0,0 +1,43 @@
+  title: string;
+}
+
+export interface MdCover {
```
this is the second MDCover interface I've come across and they both have the same properties. Maybe just keep one and export when needed

LesediMoko commented on src/models/topComic.d.ts:
```diff
@@ -0,0 +1,330 @@
+  b2key: string;
+}
+
+export interface ExtendedNew {
```
News and ExtendedNews have the same properties. Consider only keeping one

LesediMoko commented on src/utilities/api.ts:
consider adding a return type to your functions so that typescript doesn't infer them as 'any'.

LesediMoko reviewed (commented):
It all looks good. Just made a few comments regarding to the redundancy of your types as well as having a return type for your GET functions

tanadendere committed:
feat: :twisted_rightwards_arrows: Merge branch 'main' into utilities/api-calls

Created base outline of components so to get better clarity on the api return types

tanadendere committed:
feat: :fire: Cleaned up repetitive types in the interfaces

LesediMoko reviewed (changes requested):
Great job shaking off the repetitive interface, the code looks really good now! All that's missing are the return types in api.ts for your fetch responses and you'll be good to go 🏄🏽‍♂️

i.e. ` export function getComicInfo(slug: string) : Promise<IComicData> {...} ` 
or just ` export function getComicInfo(slug: string) : IComicData {...} ` if the promise is resolved within the function

tanadendere commented:
Thanks Lesedi, i had planned to add the add the return types in the next PR because I plan on implementing the API calls a bit differently.

LesediMoko reviewed (approved):
All my concerns have been addressed. Great work, Tana!
